### PR TITLE
Add host variables for advanced install

### PR DESCRIPTION
**Pull request by bfallonf**
Edits to the advanced installation section as per BZ 1245023.

Basically, edited the Configuring Ansible section to be more accurate and adding the missing information (labels and en vars) for more context.

@detiber I'll ping you in the BZ, but if there's anything you feel missing or not clear, let me know.

### CODE REVIEW

File: admin_guide/install/advanced_install.adoc [hunk 1]
```diff
@@ -83,22 +83,44 @@ the remaining instructions.
...
+link:../../architecture/infrastructure_components/kubernetes_infrastructure.html#high-availability-masters[using multiple
 masters for high availability]. You can choose an example that matches your
-requirements to modify and use as your inventory file when
-link:#running-the-ansible-installer[running the Ansible installer].
+requirements and use as your inventory file when
+link:#running-the-ansible-installer[running the Ansible installer], or modify an example to match your own environment.
+
+*Configuring Labels During the Ansible Inatall*
+
+To assign labels to hosts during the Ansible install, indicate the desired labels in the *_/etc/ansible/hosts_* file:
```
Comment (adellape) [inline admin_guide/install/advanced_install.adoc]: Suggestions:

Should be more specific that the labels are being applied to the nodes, versus generic "hosts".

Also could use a link to what labels are (https://docs.openshift.com/enterprise/3.0/architecture/core_concepts/pods_and_services.html#labels) and some context (or a link to context) on why they would care to label a node (for example, to use w/ the scheduler https://docs.openshift.com/enterprise/3.0/admin_guide/scheduler.html)

File: admin_guide/install/advanced_install.adoc [hunk 2]
```diff
@@ -83,22 +83,44 @@ the remaining instructions.
...
+The following sections describe example *_/etc/ansible/hosts_* files for various environment
+topographies, including example configurations for
+link:../../architecture/infrastructure_components/kubernetes_infrastructure.html#high-availability-masters[using multiple
 masters for high availability]. You can choose an example that matches your
-requirements to modify and use as your inventory file when
-link:#running-the-ansible-installer[running the Ansible installer].
+requirements and use as your inventory file when
+link:#running-the-ansible-installer[running the Ansible installer], or modify an example to match your own environment.
+
+*Configuring Labels During the Ansible Inatall*
```
Comment (adellape) [inline admin_guide/install/advanced_install.adoc]: "Inatall"

**Comment by adellape** [comment]
Suggest dropping "During the Ansible Install" from both of the new headings, in favor of "During the Advanced Installation", or just dropping all together (e.g., "Configuring Labels on Node Hosts" and "Configuring Host Variables").

File: admin_guide/install/advanced_install.adoc (hunk 1 above)
Comment (bfallonf) [inline admin_guide/install/advanced_install.adoc]: Had that originally, but wasn't sure if could work for masters too. Mack to "node hosts" now.

And yes, done.

File: admin_guide/install/advanced_install.adoc (hunk 2 above)
Comment (bfallonf) [inline admin_guide/install/advanced_install.adoc]: :fearful:

**Comment by bfallonf** [comment]
Changed to "Configuring Node Host Labels" and Configuring Host Variables".

@adellape Thanks for the suggestions. Let me know if there's anything else.

File: admin_guide/install/advanced_install.adoc [hunk 3]
```diff
@@ -83,22 +83,46 @@ the remaining instructions.
...
+*Configuring Node Host Labels*
+
+To assign link:../../architecture/core_concepts/pods_and_services.html#labels[labels] to node hosts during the Ansible install, indicate the desired labels in the *_/etc/ansible/hosts_* file:
+
+----
+[nodes]
+node1.example.com openshift_node_labels="{'region': 'primary', 'zone': 'east'}"
+----
+
+Labels are useful for determining the placement of pods onto nodes using the link:https:../admin_guide/scheduler.html#configurable-predicates[scheduler].
```
Comment (adellape) [inline admin_guide/install/advanced_install.adoc]: Bad link (doesn't need `admin_guide`).

**Comment by bfallonf** [comment]
@adellape Should be good now.

**Comment by detiber** [comment]
@bfallonf :+1: from me, thanks for adding the variables

File: admin_guide/install/advanced_install.adoc [hunk 4]
```diff
@@ -83,22 +83,71 @@ the remaining instructions.
...
+link:../../architecture/infrastructure_components/kubernetes_infrastructure.html#high-availability-masters[using multiple
 masters for high availability]. You can choose an example that matches your
-requirements to modify and use as your inventory file when
-link:#running-the-ansible-installer[running the Ansible installer].
+requirements and use as your inventory file when
+link:#running-the-ansible-installer[running the Ansible installer], or modify an example to match your own environment.
+
+*Configuring Node Host Labels*
+
+To assign link:../../architecture/core_concepts/pods_and_services.html#labels[labels] to node hosts during the Ansible install, indicate the desired labels in the *_/etc/ansible/hosts_* file:
```
Comment (adellape) [inline admin_guide/install/advanced_install.adoc]: (Suggestion) May be overkill, but possible rework to be more explicit and provide explanation first:

"You can assign labels to node hosts during the Ansible install using the the /etc/ansible/hosts file. Labeling nodes is useful for determining the placement of pods onto nodes using the scheduler.

To do so, use the openshift_node_labels variable to indicate the desired labels and add it to the end of the desired host entry in the [nodes] section:"

File: admin_guide/install/advanced_install.adoc [hunk 5]
```diff
@@ -83,22 +83,71 @@ the remaining instructions.
...
+
+----
+[nodes]
+node1.example.com openshift_node_labels="{'region': 'primary', 'zone': 'east'}"
+----
```
Comment (adellape) [inline admin_guide/install/advanced_install.adoc]: Should be wrapped in an example block, I think.

File: admin_guide/install/advanced_install.adoc [hunk 6]
```diff
@@ -83,22 +83,71 @@ the remaining instructions.
...
+
+*Configuring Host Variables*
+
+To assign environment variables to hosts during the Ansible install, indicate the desired variables in the *_/etc/ansible/hosts_* file:
+
+----
+[masters]
+ec2-52-6-179-239.compute-1.amazonaws.com
+openshift_public_hostname=ose3-master.public.example.com
+----
```
Comment (adellape) [inline admin_guide/install/advanced_install.adoc]: Example block.

File: admin_guide/install/advanced_install.adoc [hunk 7]
```diff
@@ -83,22 +83,71 @@ the remaining instructions.
...
+Labels are useful for determining the placement of pods onto nodes using the link:../scheduler.html#configurable-predicates[scheduler].
+
+*Configuring Host Variables*
+
+To assign environment variables to hosts during the Ansible install, indicate the desired variables in the *_/etc/ansible/hosts_* file:
+
+----
+[masters]
+ec2-52-6-179-239.compute-1.amazonaws.com
+openshift_public_hostname=ose3-master.public.example.com
```
Comment (adellape) [inline admin_guide/install/advanced_install.adoc]: I assume any variables need to stay on the same line as the host entry. ^ @detiber @sdodson ?

File: admin_guide/install/advanced_install.adoc [hunk 8]
```diff
@@ -83,22 +83,71 @@ the remaining instructions.
...
+----
+
+The following table describes example environment variables to replace
+in the above example configuration:
```
Comment (adellape) [inline admin_guide/install/advanced_install.adoc]: Seems like it's more describing common environment variables for use with the installer specifically, versus "example environment variables".

File: admin_guide/install/advanced_install.adoc [hunk 9]
```diff
@@ -83,22 +83,71 @@ the remaining instructions.
...
+
+[options="header"]
+|===
+
+|Variable |Purpose
+
+|`*openshift_hostname*`
+|This variable overrides the internal cluster hostname for the system.
```
Comment (adellape) [inline admin_guide/install/advanced_install.adoc]: s/hostname/host name/

^ In prose. A couple more below as well.

File: admin_guide/install/advanced_install.adoc [hunk 10]
```diff
@@ -83,22 +83,71 @@ the remaining instructions.
...
+Use this when the system's default IP address does not resolve to the system
+hostname.
+
+|`*openshift_public_hostname*`
+|This variable overrides the system's external/public hostname. Use this for
```
Comment (adellape) [inline admin_guide/install/advanced_install.adoc]: "external or public" ?

File: admin_guide/install/advanced_install.adoc [hunk 11]
```diff
@@ -83,22 +83,71 @@ the remaining instructions.
...
+cloud installations, or when using a Network address translation (NAT).
```
Comment (adellape) [inline admin_guide/install/advanced_install.adoc]: Seems weird to capitalize "Network" (another below as well). Not sure we need to spell it out though. Possible rewording though: "or for hosts on networks using network address translation (NAT)."

File: admin_guide/install/advanced_install.adoc (hunk 7 above)
Comment (detiber) [inline admin_guide/install/advanced_install.adoc]: @adellape that is correct, variables in the hostfile (ini) format when defined for a host need to be on the same line as  the host definition.

They can only be separated out on separate lines under a group vars heading [ose3:vars] for example.

**Comment by adellape** [comment]
@bfallonf Thanks. Couple fixes and a suggestion or two inline. :koala:

File: admin_guide/install/advanced_install.adoc (hunk 7 above)
Comment (bfallonf) [inline admin_guide/install/advanced_install.adoc]: Hmm it should be. Seems to be how it's marked up in the example... Adding a \ to the end of the line won't work in this context, correct?

**Comment by bfallonf** [comment]
@adellape Addressed, except for the one about the example being over two lines. Thanks! :cookie:

File: admin_guide/install/advanced_install.adoc (hunk 7 above)
Comment (adellape) [inline admin_guide/install/advanced_install.adoc]: Correct, they need to be on the same line.

Comment (adellape) [inline admin_guide/install/advanced_install.adoc]: Which I see they are now :+1:

**Pull request merged**